.github/workflows/docs-preview-on-demand.yaml

```
@@ -0,0 +1,173 @@
+          name: webapp
+          path: ./webapp
+          run-id: ${{ steps.find-build.outputs.run_id }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Place webapp under /client/
+        run: |
+          mkdir -p ./docs/build/current/client
+          cp -r ./webapp/. ./docs/build/current/client/
+
+      - name: Deploy preview to gh-pages
+        uses: peaceiris/actions-gh-pages@v3
```

actionlint (1.7.12)

[error] the runner of "peaceiris/actions-gh-pages@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)